ACHDictionary.go

```diff
@@ -294,22 +295,26 @@ func ACHParticipantPostalCodeFilter(achParticipants []*ACHParticipant, s string)
 	return nsl
 }
 
-// ToDo:  Use for filtering by Routing Number after a search is returned
-
-/*// ACHParticipantRoutingNumberFilter filters ACHParticipant by Routing Number
-func ACHParticipantRoutingNumberFilter(achParticipants []*ACHParticipant, s string) []*ACHParticipant {
+// ACHParticipantRoutingNumberFilter filters ACHParticipant by Routing Number
+func (f ACHDictionary) ACHParticipantRoutingNumberFilter(achParticipants []*ACHParticipant, s string) ([]*ACHParticipant, error) {
 	s = strings.TrimSpace(s)
+
+	if len(s) < MinimumRoutingNumberDigits {
+		// The first 2 digits (characters) are required
+		f.errors.Add(NewRecordWrongLengthErr(2, len(s)))
+		return nil, f.errors
+	}
 	nsl := make([]*ACHParticipant, 0)
 	for _, achP := range achParticipants {
 		if strings.HasPrefix(achP.RoutingNumber, s) {
 			nsl = append(nsl, achP)
 		}
 	}
-	return nsl
-}*/
+	return nsl, nil
+}
 
-// ACHDictionaryStateFilter filters ACHDictionary.ACHParticipant by state
-func (f *ACHDictionary) ACHDictionaryStateFilter(s string) []*ACHParticipant {
+// StateFilter filters ACHDictionary.ACHParticipant by state
+func (f ACHDictionary) StateFilter(s string) []*ACHParticipant {
```

Comment:
I chatted with Brooke about going back to pointer receivers to avoid copies and share a global instance for these methods.